src/openforms/formio/rendering/utils.py

```diff
@@ -0,0 +1,53 @@
+from typing import TYPE_CHECKING
+
+from glom import Assign, Path, glom
+
+from openforms.submissions.rendering import Renderer, RenderModes
+from openforms.submissions.rendering.nodes import SubmissionStepNode
+from openforms.typing import JSONObject
+
+from .default import ColumnsNode, EditGridGroupNode, EditGridNode, FieldSetNode
+from .nodes import ComponentNode
+
+if TYPE_CHECKING:
+    from openforms.submissions.models import Submission
+
+
+def render_nested_json(submission: "Submission") -> JSONObject:
```

Comment:
maybe name the python module `structured.py` instead of the too-generic `utils.py`?

I also find the `render_nested_json` a bit of a misnomer because it outputs python data structures suitable for json serialization, so maybe `openforms.formio.rendering.structured.render` covers the meaning better?

Reply:
It's true, although I wanted to convey that this can be used to render 'json' (python data suitable for json serialization).

Maybe `openforms.formio.rendering.structured.render_json`?

Reply:
yeah, it's a broader problem where "json" means different things in the python world, so that's a good compromise :P